### DISCUSSION TITLE
moose compiling issue on HPC

### DISCUSSION THREAD

**Discussion by rui-hu** [idaholab/moose#17146] [Q&A Getting Started] [answered]
Because of the c++17 requirements, I'm trying to use miniconda moose environment to compile moose and a moose-based app on a Linux cluster. 
The code compiling went fine, but cannot run, with the error:
_symbol lookup error: /home/cluster2/rhu/miniconda3/envs/moose/lib/./libmpifort.so.12: undefined symbol: MPII_F_TRUE_

Has anyone run into similar issues before? It seems MPI related.

**Comment by milljm** [comment, marked as answer]
It sounds like MPI supplied by miniconda is in conflict with MPI being made available elsewhere (system modules). Can you report the libraries your binary linked to?

```bash
ldd <binary>-opt
```

This might help us determine where the offending link is coming from.

**Reply by ngrilli** [reply]
Dear @rui-hu 

What is the operating system?
I suggest you try the manual installation GCC/MPICH
which has always been successful on our CentOS 7

Best Regards,
Nicolò Grilli
National University of Singapore

**Reply by rui-hu** [reply]
Thank you both. It is indeed an issue that miniconda is in conflict with the mpi loaded by the system. 

"module unload" the system mpi solved the problem. 

For those interested, the cluster is on CentOS.
